joplin/base/models/janis_page.py

```diff
@@ -43,32 +43,50 @@ class JanisBasePage(Page):
     )
 
     def janis_url(self):
+        """
+        This function parses various attributes of content types to construct the
+        expected url structure for janis
+
+        It probably could use some refactoring.
+        """
         page_slug = self.slug
 
         if self.janis_url_page_type == "department":
             return os.environ["JANIS_URL"] + "/en/" + page_slug
 
         if self.janis_url_page_type == "topiccollection":
-            theme_slug = self.theme.slug;
-            return os.environ["JANIS_URL"] + "/en/" + theme_slug + "/" + page_slug
+            try:
+                theme_slug = self.theme.slug
+                return os.environ["JANIS_URL"] + "/en/" + theme_slug + "/" + page_slug
+            except Exception as e:
```

Comment:
its mostly NoneType, Value, or Attribute errors. 

But in this case a think a general exception is ok.

Reply:
This code isn't handling errors so much as it is suppressing errors. Would we want to return a null/default `janis_url` if the original url construction fails? I guess it's okay since coming up a complete solution is documented in another issue.

Reply:
nvm. The complete solution is literally the next PR